src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs

```diff
@@ -130,18 +136,48 @@ public static async Task<Solution> AddParameterAsync(
                 }
 
                 if (method.MethodKind == MethodKind.ReducedExtension && insertionIndex < existingParameters.Count)
-                {
                     insertionIndex++;
+
+                var memberToAssignTo = await GetMemberToAssignToAsync(semanticDocument).ConfigureAwait(false);
+                if (memberToAssignTo != null)
+                {
+                    var initializeParameterService = documentLookup.Key.GetRequiredLanguageService<IInitializeParameterService>();
+                    var body = initializeParameterService.GetBody(methodNode);
+                    if (semanticDocument.SemanticModel.GetOperation(body, cancellationToken) is IBlockOperation blockOperation)
+                    {
+                        initializeParameterService.AddAssignment(
+                            methodNode, blockOperation, parameterSymbol, memberToAssignTo, editor);
+                    }
                 }
 
-                AddParameterEditor.AddParameter(syntaxFacts, editor, methodNode, insertionIndex, parameterDeclaration, cancellationToken);
+                AddParameterEditor.AddParameter(
+                    syntaxFacts, editor, methodNode, insertionIndex, parameterDeclaration, cancellationToken);
             }
 
             var newRoot = editor.GetChangedRoot();
             solution = solution.WithDocumentSyntaxRoot(document.Id, newRoot);
         }
 
         return solution;
+
+        async Task<ISymbol?> GetMemberToAssignToAsync(SemanticDocument document)
+        {
+            if (method.MethodKind != MethodKind.Constructor)
+                return null;
+
+            if (argument is null)
+                return null;
+
+            var (_, parameterToExistingMember, _, _) = await GenerateConstructorHelpers.GetParametersAsync(
```

Comment:
we don't need the other values returned here.  just the computation about which existing fields we could map to.